### PR TITLE
[mtl] Blit image types

### PR DESCRIPTION
**Pull request by kvark**
Follow-up to #1982
~~Base on WIP work of #1995~~

PR checklist:
- [x] `make` succeeds (on *nix)
- [x] `make reftests` succeeds
- [x] tested examples with the following backends:

TODO:
- [x] block on https://github.com/gfx-rs/metal-rs/pull/45
~~- [ ] wait for #1995~~
- [x] rebase
- [ ] review (can be done now)

### CODE REVIEW

File: src/backend/metal/shaders/commands.metal [hunk 1]
```diff
@@ -18,18 +18,59 @@ vertex VertexData vs_blit(TextureBlitAttributes in [[stage_in]]) {
...
-fragment float4 ps_blit_array(
+fragment float4 ps_blit_2d_array(
     VertexData in [[stage_in]],
     texture2d_array<float> tex2DArray [[ texture(0) ]],
     sampler sampler2D [[ sampler(0) ]]
 ) {
   return tex2DArray.sample(sampler2D, in.uv.xy, uint(in.uv.z), level(in.uv.w));
 }
+
+fragment float4 ps_blit_2d_ms(
```
Comment (msiglreith) [inline src/backend/metal/shaders/commands.metal]: uhm, blit doesn'T support multisampled images

Comment (kvark) [reply src/backend/metal/shaders/commands.metal]: heh, ok. I was thinking about GL when adding this. Will remove.

**Comment by kvark** [comment]
@grovesNL please review

**Comment by kvark** [comment]
Just didn't get my hands on it, going to check the specs and add this now...

**Comment by kvark** [comment]
Actually, it's not clear to me how Cube blitting is going to work at all. We can follow-up later.
bors r=grovesNL

**Pull request merged**